File: windows/Models/MsuTrackCatalog.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Text.Json;

namespace LTTPRandomizerGenerator.Models;

public static class MsuTrackCatalog
{
    private record CatalogEntry(int Slot, string Name, string Type);

    public static List<MsuTrackSlot> Load()
    {
        var assembly = Assembly.GetExecutingAssembly();
        var resourceName = assembly.GetManifestResourceNames()
            .First(n => n.EndsWith("trackCatalog.json", StringComparison.OrdinalIgnoreCase));

        using var stream = assembly.GetManifestResourceStream(resourceName)!;
        using var reader = new StreamReader(stream);
        var json = reader.ReadToEnd();

        var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
        var entries = JsonSerializer.Deserialize<List<CatalogEntry>>(json, options)!;

        return entries.Select(e => new MsuTrackSlot
        {
            SlotNumber = e.Slot,
            Name = e.Name,
            TrackType = e.Type
        }).ToList();
    }
}

File: android/app/src/main/java/com/lttprandomizer/MsuTrackCatalog.kt
package com.lttprandomizer

import android.content.Context
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json

object MsuTrackCatalog {
    @Serializable
    private data class CatalogEntry(val slot: Int, val name: String, val type: String)

    fun load(context: Context): List<MsuTrackSlot> {
        val json = context.resources.openRawResource(R.raw.track_catalog)
            .bufferedReader().use { it.readText() }
        val entries = Json.decodeFromString<List<CatalogEntry>>(json)
        return entries.map { MsuTrackSlot(slotNumber = it.slot, name = it.name, trackType = it.type) }
    }
}
